packages/typespec-ts/src/utils/credentialUtils.ts

```diff
@@ -39,3 +40,44 @@ export function getSupportedHttpAuth(
 
   return authList;
 }
+
+export function isSupportedKeyCredential(auth: HttpAuth): boolean {
+  return (
+    (auth.type === "apiKey" && auth.in === "header") || auth.type === "http"
+  );
+}
+
+export function isSupportedTokenCredential(auth: HttpAuth): boolean {
+  return auth.type === "oauth2" || auth.type === "openIdConnect";
+}
+
+export function hasKeyCredential(initialization: SdkInitializationType) {
+  const authScheme = getAuthScheme(initialization);
+  return authScheme.some((auth) => isSupportedKeyCredential(auth));
+}
+
+export function hasTokenCredential(initialization: SdkInitializationType) {
+  const authScheme = getAuthScheme(initialization);
+  return authScheme.some((auth) => isSupportedTokenCredential(auth));
+}
+
+function getAuthScheme(initialization: SdkInitializationType): HttpAuth[] {
+  const credentialParams = initialization.properties?.find(
+    (param) => param.kind === "credential"
+  );
+  if (!credentialParams) {
+    return [];
+  }
+  const kind = credentialParams.type.kind;
+  const authScheme: HttpAuth[] = [];
+  if (kind === "credential") {
+    authScheme.push(credentialParams.type.scheme);
+  } else if (kind === "union") {
+    for (const param of credentialParams.type.values) {
+      if (param.kind === "credential") {
```

Comment:
i don't think this is possible. we didn't do any casting here because the tcgc type is only allowed to be credential.

https://github.com/Azure/typespec-azure/blob/main/packages/typespec-client-generator-core/src/interfaces.ts#L427-L431

Reply:
let's remove this ===